### ISSUE TITLE
Incorrect transclusion errors for zero selector ng-content

### ISSUE THREAD

**Issue by bencalabrese** [dart-lang/angular_analyzer_plugin#480]
I'm seeing the following incorrect validation error in VS Code:
"The content does not match any transclusion selectors of the surrounding component"

This is happening when including any content in a [MaterialButtonComponent](https://github.com/dart-lang/angular_components/tree/master/lib/material_button).  Docs seem to indicate that this error would not be exposed since MaterialButtonComponent doesn't have a selector specified for its ng-content tag.

This also appears to be happening for [MaterialDialogComponent](https://github.com/dart-lang/angular_components/tree/master/lib/material_dialog) which has some a couple ng-content tags with selectors as well as one without. In that instance, all tags included inside the MaterialDialogComponent are getting the error, even if they include one of the specified selectors (like [header]).

```html
<material-dialog>
  <div header></div>
  <div></div>         <----- All three inner tags show errors, but all should be valid.
  <div footer></div>
</material-dialog>
```

**Comment by MichaelRFairhurst** [comment]
This is unfortunately some kind of race condition/cache corruption issue that I haven't figured out yet.

You should be able to fix it by clearing out `~/.dartServer/.analysis-driver/*.ng*` and restarting your analysis server, for now.

**Comment by bencalabrese** [comment]
Sounds good. Clearing out that cache seems to have worked for the time being. Thanks for taking a look!

**Comment by MichaelRFairhurst** [comment]
I worked on possible causes of race conditions for quite a bit last quarter, and haven't seen this since. I'm closing but please do file again if it comes back!

**Comment by bencalabrese** [comment]
I saw this earlier today :-/

Sorry, I've lost my repro at this point, but it is still happening.

**Comment by drbcode** [comment]
I ran into this issue as well, this issue was resolved after deleting `~/.dartServer/.analysis-driver/`

I guess this issue can be reproduced on material dialog example if directives and providers are not specified.

```
warning: The content does not match any transclusion selectors of the surrounding component (content_not_transcluded at [crx_session_switcher] lib/src/ng/comp/session_info/session_info.html:64)
```
